scanpy/plotting/_tools/scatterplots.py

```diff
@@ -262,6 +262,7 @@ def embedding(
     #     color=gene2, components = [1, 2], color=gene2, components=[2,3],
     # ]
     for count, (value_to_plot, dims) in enumerate(zip(color, dimensions)):
+        kwargs = kwargs_scatter.copy()  # is potentially mutated for each plot
```

Comment:
Maybe I was not clear, but I mean to not have some object outside the loop.
```suggestion
        kwargs_scatter = kwargs.copy()  # is potentially mutated for each plot
```
And the replace all `kwargs` in the loop with `kwargs_scatter` since they are only used on `scatter` and then make `kwargs_outline` from `kwargs_scatter`

Reply:
Is there something I'm missing about this?  I think defining a variable to store something outside a loop is a little strange.

Reply:
No, that makes total sense. I basically did what you want, but less clean and with a smaller diff.

In the long run, we should just extract the loop body to a function and only pass a copy into it, then all is good.